test/calibration/test_calibrations.py

```diff
@@ -1329,17 +1329,23 @@ def test_save_load_parameter_values(self):
 
         # Check that we cannot rewrite files as they already exist.
         with self.assertRaises(CalibrationError):
-            self.cals.save("csv")
+            self.cals.save("csv", file_prefix=prefix)
+
+        self.cals.save("csv", overwrite=True, file_prefix=prefix)
 
-        self.cals.save("csv", overwrite=True)
+        self._remove_files(prefix)
```

Comment:
Can we make this `self.addCleanup(self._remove_files, prefix)` but move it up to right after the first save call. That way the test will delete the files if the test fails or passes. Right now doing it in the test code directly means we only delete the files if everything passes and the test reaches the end

Reply:
Done here: 246ea96